fence/agents/powerman/fence_powerman.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python
+import os
+import time
+from datetime import datetime
+import sys
+import subprocess
+import re
+import atexit
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from fencing import *
+from fencing import run_delay
+import logging
+
+
+#BEGIN_VERSION_GENERATION
+RELEASE_VERSION="Powerman Fencing Agent"
+REDHAT_COPYRIGHT=""
+BUILD_DATE=""
+#END_VERSION_GENERATION
+
+
+#### important!!! #######
+class PowerMan:
+	"""Python wrapper for calling powerman commands
+
+	This class makes calls to a powerman deamon for a cluster of computers.
+	The make-up of such a call looks something like:
+		$ pm -h elssd1:10101 <option> <node>
+	where option is something like --off, --on, --cycle and where node is 
+	elssd8, or whatever values are setup in powerman.conf (***this is key, 
+	because otherwise this code will not work!)
+	"""
+	program_name = "powerman"
+
+	def __init__(self, server_name, port="10101"):
```

Comment:
magic number 10101 is not required here because you have defined default ipport so either it will be defined by user or default value will be used.